src/content/scripting/script_names.h

```diff
@@ -31,23 +31,23 @@
 #include "cds_objects.h"
 #include "metadata/metadata_handler.h"
 
-static constexpr auto res_names = std::array<std::pair<resource_attributes_t, const char*>, R_MAX> {
-    std::pair(R_SIZE, "R_SIZE"),
-    std::pair(R_DURATION, "R_DURATION"),
-    std::pair(R_BITRATE, "R_BITRATE"),
-    std::pair(R_SAMPLEFREQUENCY, "R_SAMPLEFREQUENCY"),
-    std::pair(R_NRAUDIOCHANNELS, "R_NRAUDIOCHANNELS"),
-    std::pair(R_RESOLUTION, "R_RESOLUTION"),
-    std::pair(R_COLORDEPTH, "R_COLORDEPTH"),
-    std::pair(R_PROTOCOLINFO, "R_PROTOCOLINFO"),
-    std::pair(R_RESOURCE_FILE, "R_RESOURCE_FILE"),
-    std::pair(R_BITS_PER_SAMPLE, "R_BITS_PER_SAMPLE"),
-    std::pair(R_TYPE, "R_TYPE"),
-    std::pair(R_FANART_OBJ_ID, "R_FANART_OBJ_ID"),
-    std::pair(R_FANART_RES_ID, "R_FANART_RES_ID"),
-    std::pair(R_LANGUAGE, "R_LANGUAGE"),
-    std::pair(R_AUDIOCODEC, "R_AUDIOCODEC"),
-    std::pair(R_VIDEOCODEC, "R_VIDEOCODEC"),
+static auto res_names = std::map<ResourceAttribute, std::string_view> {
```

Comment:
hmm?

Reply:
Maps are great and fit what we want to achieve here, e.g. lookup by value. We dont need the constexpr stuff because we dont do any compile time computation with these maps, they are only queried at runtime, so static does the job here.